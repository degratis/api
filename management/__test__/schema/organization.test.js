const { createTestClient } = require('apollo-server-testing');
const { gql, ApolloServer } = require("apollo-server")

const { resolvers, typeDefs } = require("../../schema")

const mocks = {
  ID: () => "123",
  Status: () => "PENDING",
  DateTime: () => new Date('2021-05-01'),
}

const server = new ApolloServer({ resolvers, typeDefs, mocks });
const { query } = createTestClient(server);

const LIST_ORGANIZATIONS = gql`
  query ListOrganizations($nextToken: String, $limit: Int) {
    listOrganizations(nextToken: $nextToken, limit: $limit) {
      items {
        id
        name
        slug
        status
        createdAt
        updatedAt
      }
      nextToken
    }
  }
`

const GET_ORGANIZATION = gql`
  query GetOrganization($id: ID!) {
    getOrganization(id: $id) {
      id
      name
      slug
      status
      createdAt
      updatedAt
    }
  }
`

describe('Organization', () => {
  describe('Query', () => {
    it('should list organizations', async () => {
      const result = await query({
        query: LIST_ORGANIZATIONS,
      });

      expect(result).toMatchSnapshot()
    })
  })

  it('should get one organization', async () => {
    const result = await query({
      query: GET_ORGANIZATION,
      variables: { id: 1 }
    });

    expect(result).toMatchSnapshot()
  })
})
