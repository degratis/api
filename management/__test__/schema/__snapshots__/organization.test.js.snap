// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization Query should list organizations 1`] = `
Object {
  "data": Object {
    "listOrganizations": Object {
      "items": Array [
        Object {
          "createdAt": 2021-05-01T00:00:00.000Z,
          "id": "123",
          "name": "Hello World",
          "slug": "Hello World",
          "status": "PENDING",
          "updatedAt": 2021-05-01T00:00:00.000Z,
        },
        Object {
          "createdAt": 2021-05-01T00:00:00.000Z,
          "id": "123",
          "name": "Hello World",
          "slug": "Hello World",
          "status": "PENDING",
          "updatedAt": 2021-05-01T00:00:00.000Z,
        },
      ],
      "nextToken": "Hello World",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Organization should get one organization 1`] = `
Object {
  "data": Object {
    "getOrganization": Object {
      "createdAt": 2021-05-01T00:00:00.000Z,
      "id": "123",
      "name": "Hello World",
      "slug": "Hello World",
      "status": "PENDING",
      "updatedAt": 2021-05-01T00:00:00.000Z,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
